core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-box/template-builder-box.component.spec.ts

```diff
@@ -181,4 +186,21 @@ describe('TemplateBuilderBoxComponent', () => {
 
         expect(rejectDeleteMock).toHaveBeenCalled();
     });
+
+    it('should use titles from container map', (done) => {
+        const displayedContainerTitles = spectator
+            .queryAll(byTestId('container-title'))
+            .map((element) => element.textContent.trim());
+        const containerMapTitles = Object.values(CONTAINER_MAP_MOCK).map(
+            (container) => container.title
+        );
+
+        displayedContainerTitles.forEach((title) => {
+            if (!containerMapTitles.includes(title)) {
+                throw new Error(`title: "${title} not included the container map is displayed`);
+            }
+        });
```

Comment:
Normally, our tests do `expect(xxxx).toBe(xxxx)` we never do tests without expect, and I am not certain how `throw new Error` affects or works in the tests.

Can you explain this approach? or put some links that we can understand this.

Reply:
I was thinking on writing some kind of check where we could expect the displayed titles to be a subset of the containerMapTitles, however we would have had to write an is subset function, which could be a valid approach. However, this other one also works, I verified. One interesting thing is that I was used to using a `fail` function from jest which I was able to use locally, but wasn't supported in the CI/CD.

My source was this SO answer: [https://stackoverflow.com/a/48196284](https://stackoverflow.com/a/48196284)

Reply:
TIL! Kind of cool approach to make the test fail, not sure how this shows in the results.

Altho, the error message is weird. I am not sure if "is displayed" make sense here.

Reply:
@zJaaal @rjvelazco @hmoreras @manuelrojas @oidacra Take a look at this approach.

Reply:
This is really cool. But is there a way to expect it to not throw an error? Just to make it cleaner and clearer

Reply:
@zJaaal, that is done by using the done callback as it is happening.